### PR TITLE
Adding some more tests for Microsoft.Repl

### PR DESCRIPTION
**Pull request by tlmii**
Adding tests for AnsiColorExtensions, Disposable, Disposable<T> and some of CoreParseResult. Also removing some unused code from the Point class.

### CODE REVIEW

File: src/Microsoft.Repl.Tests/ConsoleHandling/AnsiColorExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        [Theory]
+        [InlineData("BlackText", AllowedColors.BoldBlack, "\x1B[30m\x1B[1mBlackText\x1B[22m\x1B[39m")]
+        [InlineData("RedText", AllowedColors.BoldRed, "\x1B[31m\x1B[1mRedText\x1B[22m\x1B[39m")]
+        [InlineData("GreenText", AllowedColors.BoldGreen, "\x1B[32m\x1B[1mGreenText\x1B[22m\x1B[39m")]
+        [InlineData("YellowText", AllowedColors.BoldYellow, "\x1B[33m\x1B[1mYellowText\x1B[22m\x1B[39m")]
+        [InlineData("BlueText", AllowedColors.BoldBlue, "\x1B[34m\x1B[1mBlueText\x1B[22m\x1B[39m")]
+        [InlineData("MagentaText", AllowedColors.BoldMagenta, "\x1B[35m\x1B[1mMagentaText\x1B[22m\x1B[39m")]
+        [InlineData("CyanText", AllowedColors.BoldCyan, "\x1B[36m\x1B[1mCyanText\x1B[22m\x1B[39m")]
+        [InlineData("WhiteText", AllowedColors.BoldWhite, "\x1B[37m\x1B[1mWhiteText\x1B[22m\x1B[39m")]
+        public void SetColor_WithBold_Both(string text, AllowedColors color, string expected)
```
Comment (bhsubra) [inline src/Microsoft.Repl.Tests/ConsoleHandling/AnsiColorExtensionsTests.cs]: nit: Can we update this and below test names? From the last part, it's not very clear what the expected output is.

File: src/Microsoft.Repl.Tests/Parsing/CoreParseResultTests.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            Assert.Single(result.SectionStartLookup);
+            Assert.Equal(0, result.SectionStartLookup.Single().Key);
+            Assert.Equal(0, result.SectionStartLookup.Single().Value);
+        }
+
+        [Theory]
+        [InlineData("set header content-type application/json", 1, "header content-type application/json")]
+        [InlineData("set header content-type application/json", 2, "content-type application/json")]
+        [InlineData("set header content-type application/json", 3, "application/json")]
+        public void Slice_CorrectCommandText(string commandText, int toRemove, string expectedCommandText)
```
Comment (bhsubra) [inline src/Microsoft.Repl.Tests/Parsing/CoreParseResultTests.cs]: nit: Can we follow this naming convention-<MethodBeingTested>_<DescribeInputScenario>_<DescribeExpectedResult>?

File: src/Microsoft.Repl.Tests/ConsoleHandling/AnsiColorExtensionsTests.cs (hunk 1 above)
Comment (tlmii) [reply src/Microsoft.Repl.Tests/ConsoleHandling/AnsiColorExtensionsTests.cs]: Done in 4a073b8.

File: src/Microsoft.Repl.Tests/Parsing/CoreParseResultTests.cs (hunk 2 above)
Comment (tlmii) [reply src/Microsoft.Repl.Tests/Parsing/CoreParseResultTests.cs]: I have a hard time following that when using the Theory and parameterized tests. The input scenario is a little harder to define. I gave it a shot in 4a073b8 though.

**Pull request merged**